src/qililab/platform/platform.py

```diff
@@ -526,6 +528,40 @@ def __str__(self) -> str:
         """
         return str(YAML().dump(self.to_dict(), io.BytesIO()))
 
+    def execute_qprogram(self, qprogram: QProgram):
+        """Execute a QProgram using the platform instruments.
+
+        Args:
+            qprogram (QProgram): The QProgram to execute.
+
+        Returns:
+            list[Result]: Result list obtained from the execution.
+        """
+        qblox_compiler = QbloxCompiler()
+        sequences = qblox_compiler.compile(qprogram=qprogram)
+        buses = {bus_alias: self._get_bus_by_alias(alias=bus_alias) for bus_alias in sequences}
+
+        # Upload sequences
+        for bus_alias in sequences:
+            buses[bus_alias].upload_qpysequence(qpysequence=sequences[bus_alias])
+
+        # Execute sequences
+        for bus_alias in sequences:
+            buses[bus_alias].run()
```

Comment:
Do we need to have 3 separated methods: upload, run, acquire?

Why not having one execute, and doing everything inside the driver?